collectors/python.d.plugin/energid/energid.chart.py

```diff
@@ -0,0 +1,150 @@
+            ['mempool_current', 'Usage', 'absolute', None, 1024*1024],
+            ['mempool_txsize', 'TX Size', 'absolute', None, 1024*1024],
+        ],
+    },
+    'secmem': {
+        'options': [None, 'Secure Memory', 'KiB', 'memory', 'energid.secmem', 'area'],
+        'lines': [
+            ['secmem_total', 'Total', 'absolute', None, 1024],
+            ['secmem_locked', 'Locked', 'absolute', None, 1024],
+            ['secmem_used', 'Used', 'absolute', None, 1024],
+        ],
+    },
+    'network': {
+        'options': [None, 'Network', 'count', 'network', 'energid.network', 'line'],
+        'lines': [
+            ['network_connections', 'Connections', 'absolute'],
+        ],
+    },
+    'timeoffset': {
+        'options': [None, 'Network', 'seconds', 'network', 'energid.timeoffset', 'line'],
+        'lines': [
+            ['network_timeoffset', 'offseet', 'absolute'],
+        ],
+    },
+    'utxo': {
+        'options': [None, 'UTXO', 'count', 'UTXO', 'energid.utxo', 'line'],
+        'lines': [
+            ['utxo_count', 'UTXO', 'absolute'],
+        ],
+    },
+    'xfers': {
+        'options': [None, 'UTXO', 'count', 'UTXO', 'energid.xfers', 'line'],
+        'lines': [
+            ['utxo_xfers', 'Xfers', 'absolute'],
+        ],
+    },
+}
+
+METHODS = {
+    'getblockchaininfo': lambda r: {
+        'blockchain_blocks': r['blocks'],
+        'blockchain_headers': r['headers'],
+        'blockchain_difficulty': r['difficulty'],
+    },
+    'getmempoolinfo': lambda r: {
+        'mempool_txcount': r['size'],
+        'mempool_txsize': r['bytes'],
+        'mempool_current': r['usage'],
+        'mempool_max': r['maxmempool'],
+    },
+    'getmemoryinfo': lambda r: dict([
+        ('secmem_' + k, v) for (k,v) in r['locked'].items()
+    ]),
+    'getnetworkinfo': lambda r: {
+        'network_timeoffset' : r['timeoffset'],
+        'network_connections': r['connections'],
+    },
+    'gettxoutsetinfo': lambda r: {
+        'utxo_count' : r['txouts'],
+        'utxo_xfers' : r['transactions'],
+        'utxo_size' : r['disk_size'],
+        'utxo_amount' : r['total_amount'],
+    },
+}
+
+class Service(UrlService):
+    def __init__(self, configuration=None, name=None):
+        UrlService.__init__(self, configuration=configuration, name=name)
+        self.order = ORDER
+        self.definitions = CHARTS
+        self.host = self.configuration.get('host', '127.0.0.1')
+        self.port = self.configuration.get('port', 9796)
+        self.url = '{scheme}://{host}:{port}'.format(
+            scheme=self.configuration.get('scheme', 'http'),
+            host=self.host,
+            port=self.port,
+        )
+        self.method = 'POST'
+        self.header = {
+            'Content-Type': 'application/json',
+        }
+
+    def _get_data(self):
+        batch = []
+
+        for i, method in enumerate(METHODS):
+            batch.append({
+                'version': '1.1',
```

Comment:
The version value 1.1 sounds like something that should exist in the context of the collector config, so that it can be set to something different if it ever changes without touching the implementation

Reply:
Oh, and it would be nice to have a link to the docs how to build body with all these versions and methods. Add it as comment please @andvgal

Reply:
A comment has been added. The version does not really matters. As the payload is created manually, I believe putting JSON-RPC version into configuration has no use at all and just bloats the code. However, it can be put into a module variable just to eat a bit more memory.

Reply:
I see. If the version does not matter, it would be cleaner if the other end has the version info as optional field and then we can simply omit the field completely. That would simplify things.

Otherwise we can stick with the variable i guess

Reply:
>  I believe putting JSON-RPC version into configuration has no use at all and just bloats the code

i agree, we can add it later if needed, i think it is ok to have it hardcoded for now

Reply:
Any way, a constant with JSON-RPC version has been added on module level.